vitrivr-engine-index/src/main/kotlin/org/vitrivr/engine/index/decode/FFmpegVideoDecoder.kt

```diff
@@ -116,27 +115,30 @@ class FFmpegVideoDecoder : DecoderFactory {
 
                 /* Create consumer. */
                 val consumer = InFlowFrameConsumer(this, sourceRetrievable)
-                val ffmpegInstance = this@Instance.ffmpeg.addInput(
-                    if (source is FileSource) {
-                        UrlInput.fromPath(source.path)
-                    } else {
-                        PipeInput.pumpFrom(source.newInputStream())
-                    }
-                ).addOutput(FrameOutput.withConsumerAlpha(consumer))
+
 
                 /* Execute. */
                 try {
-                    ffmpegInstance.execute()
-                } catch (e: JaffreeException) {
-                    logger.warn(e) { "Error while decoding source ${source.name} (${source.sourceId})." }
-                } finally {
+                    source.newInputStream().use {
+                        var output = FrameOutput.withConsumerAlpha(consumer).disableStream(StreamType.SUBTITLE).disableStream(StreamType.DATA)
+                        if (!this@Instance.video) {
+                            output = output.disableStream(StreamType.VIDEO)
+                        }
+                        if (!this@Instance.audio) {
+                            output = output.disableStream(StreamType.AUDIO)
+                        }
+                        this@Instance.ffmpeg.addInput(PipeInput.pumpFrom(it)).addOutput(output).execute()
```

Comment:
I guess it would be beneficial not to enforce the detour through a stream when random access through a file is available.